rosidl_runtime_rs/src/string.rs

```diff
@@ -224,6 +225,42 @@ macro_rules! string_impl {
             }
         }
 
+        impl From<&std::string::String> for $string {
```

Comment:
Since the existing `From<&str>` impl is more general, I think this impl isn't necessary.

Reply:
So in `*self = Self::from(&s);`, `&s` is actually of type `&String` and `From` isn't implemented for that type. So I removed this impl by just using `Self::from(s.as_str())` in the `Extend` impl

Reply:
That's correct, though using `Self::from(&s)` should work too due to [Deref coercion](https://doc.rust-lang.org/book/ch15-02-deref.html#implicit-deref-coercions-with-functions-and-methods).